src/plots/piePlot.ts

```diff
@@ -243,6 +267,89 @@ export module Plots {
       var avgAngle = (startAngle + endAngle) / 2;
       return { x: avgRadius * Math.sin(avgAngle), y: -avgRadius * Math.cos(avgAngle) };
     }
+
+    protected _additionalPaint(time: number) {
+      if (this._labelsEnabled) {
+        Utils.Window.setTimeout(() => this._drawLabels(), time);
+      }
+    }
+
+    private _sliceIndexForPoint(p: Point) {
+      var pointRadius = Math.sqrt(Math.pow(p.x, 2) + Math.pow(p.y, 2));
+      var pointAngle = Math.acos(-p.y / (1 + pointRadius));
+      if (p.x < 0) {
+        pointAngle = Math.PI * 2 - pointAngle;
+      }
+      var index: number;
+      for (var i = 0; i < this._startAngles.length; i++) {
+        if (this._startAngles[i] < pointAngle && this._endAngles[i] > pointAngle) {
+          index = i;
+          break;
+        }
+      }
+      if (index !== undefined) {
+        var dataset = this.datasets()[0];
+        var datum = dataset.data()[index];
+        var innerRadius = this.innerRadius().accessor(datum, index, dataset);
+        var outerRadius = this.outerRadius().accessor(datum, index, dataset);
+        if (pointRadius > innerRadius && pointRadius < outerRadius) {
+          return index;
+        }
+      }
+      return null;
+    }
+
+    private _drawLabels() {
+      var attrToProjector = this._generateAttrToProjector();
+      var labelArea = this._renderArea.append("g").classed("label-area", true);
+      var measurer = new SVGTypewriter.Measurers.Measurer(labelArea);
+      var writer = new SVGTypewriter.Writers.Writer(measurer);
+      var dataset = this.datasets()[0];
+
+      for (var i = 0; i < dataset.data().length; i++) {
+        var datum = dataset.data()[i];
+        var value = "" + this.sectorValue().accessor(datum, i, dataset);
+        var measurement = measurer.measure(value);
+
+        var theta = (this._endAngles[i] + this._startAngles[i]) / 2;
+        var outerRadius = this.outerRadius().accessor(datum, i, dataset);
+        if (this.outerRadius().scale) {
+          outerRadius = this.outerRadius().scale.scale(outerRadius);
+        }
+        var innerRadius = this.innerRadius().accessor(datum, i, dataset);
+        if (this.innerRadius().scale) {
+          innerRadius = this.innerRadius().scale.scale(innerRadius);
+        }
+        var labelRadius = (outerRadius + innerRadius) / 2;
+
+        var x = Math.sin(theta) * labelRadius - measurement.width / 2;
+        var y = -Math.cos(theta) * labelRadius - measurement.height / 2;
+
+        var corners = [
+          { x: x, y: y },
+          { x: x, y: y + measurement.height},
+          { x: x + measurement.width, y: y },
+          { x: x + measurement.width, y: y + measurement.height }
+        ];
+
+        var sliceIndices = corners.map((corner) => this._sliceIndexForPoint(corner));
+        var showLabel = sliceIndices.every((index) => (index != null && index === i));
```

Comment:
Just `index === i` would be enough right?

Also it's worth renaming `i` to something like `datumIndex` so the logic is clearer.